core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -163,12 +164,20 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
     }
   }
 
-   /**
-    * Choose the log to clean next and add it to the in-progress set. We recompute this
-    * each time from the full set of logs to allow logs to be dynamically added to the pool of logs
-    * the log manager maintains.
-    */
   def grabFilthiestCompactedLog(time: Time, preCleanStats: PreCleanStats = new PreCleanStats()): Option[LogToClean] = {
+    if (fineGrainedLockEnable) {
+      grabFilthiestCompactedLogWithFineGrainedLock(time, preCleanStats)
+    } else {
+      grabFilthiestCompactedLogWithLongLastingLock(time, preCleanStats)
+    }
```

Comment:
In the long term, does this need to be configurable? If so, why, and if not, can you add a TODO-style comment to clean it up once the new approach is sufficiently well tested?

Reply:
Done.